docs/development/go/go-context.md

```diff
@@ -0,0 +1,468 @@
+
+{{< note >}}
+One of the return values of `Context.Done()` is a Go channel, because otherwise you would not have
+been able to use it in a `select` statement.
+{{< /note >}}
+
+On the other hand the `cancel` variable in `f2()` comes from `context.WithTimeout()`.
+This function showcases the use of the `context.WithTimeout()` function, which requires
+two parameters: a `Context` parameter and a `time.Duration` parameter. When the timeout
+period expires, the `cancel()` function is called automatically.
+
+Last, the `cancel` variable in `f3()` comes from `context.WithDeadline()`. The `f3()` function
+illustrates the use of the `context.WithDeadline()` function that requires two parameters:
+a `Context` variable and a `time` in the future that signifies the deadline of the operation.
+When the deadline passes, the `cancel()` function is called automatically.
+
+{{< note >}}
+Notice that contexts should not be stored in structures – they should be passed as
+separate parameters to functions. It is considered a good practice to pass them as
+the first parameter of a function.
+{{< /note >}}
+
+### Using simple.go
+
+Executing `simple.go` with a delay period of *4 seconds* will generate the following kind of output:
+
+    go run simple.go 4
+{{< output >}}
+Delay: 4
+f1(): 2019-05-29 08:39:48.219616 +0300 EEST m=+4.004039510
+f2(): context deadline exceeded
+f3(): 2019-05-29 08:39:56.224505 +0300 EEST m=+12.009254674
+{{< /output >}}
+
+The long lines of the output are the return values of the `time.After()` function calls.
+They denote normal operation of the program. The point here is that the operation of the
+program is cancelled when there are delays in its execution.
+
+If you use a bigger delay (*10 seconds*), which is executed as a call to `time.Sleep()`, you
+will get the following kind of output:
+
+    go run simple.go 10
+{{< output >}}
+Delay: 10
+f1(): context canceled
+f2(): context canceled
+f3(): context canceled
+{{< /output >}}
+
+The calls to `time.Sleep()` simulate a program that is slow or an operation that takes too
+much time to finish. Production code does not usually have such `time.Sleep()` function calls.
+
+## Using Context for HTTP
+
+In this section of the guide you will learn how to timeout HTTP connections on the client side.
+
+The presented utility, which is called `http.go`, requires two command line arguments,
+which are the URL to connect to and the allowed delay value in seconds.
+
+### Explaining the Go Code of the Example
+
+The Go code of the `http.go` utility is the following:
+
+{{< file "./http.go" go >}}
+package main
+
+import (
+        "context"
+        "fmt"
+        "io/ioutil"
+        "net/http"
+        "os"
+        "strconv"
+        "sync"
+        "time"
+)
+
+var (
+        myUrl string
+        delay int = 5
+        w     sync.WaitGroup
+)
+
+type myData struct {
+        r   *http.Response
+        err error
+}
+
+func connect(c context.Context) error {
```

Comment:
Now might be a good time to highlight that in packages that use contexts, convention is to pass them as the first argument to a function.

Reply:
Done.